pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -230,17 +230,22 @@ func (s *DiscoveryServer) distributedVersions(w http.ResponseWriter, req *http.R
 		var results []SyncedVersions
 		adsClientsMutex.RLock()
 		for _, con := range adsClients {
-			con.mu.RLock()
-			if con.node != nil && (proxyNamespace == "" || proxyNamespace == con.node.ConfigNamespace) {
-				// TODO: handle skipped nodes
-				results = append(results, SyncedVersions{
-					ProxyID:         con.node.ID,
-					ClusterVersion:  s.getResourceVersion(con.ClusterNonceAcked, resourceID, knownVersions),
-					ListenerVersion: s.getResourceVersion(con.ListenerNonceAcked, resourceID, knownVersions),
-					RouteVersion:    s.getResourceVersion(con.RouteNonceAcked, resourceID, knownVersions),
-				})
-			}
-			con.mu.RUnlock()
+			// wrap this in independent scope so that panic's don't bypass Unlock...
+			func() {
+				con.mu.RLock()
+				defer con.mu.RUnlock()
```

Comment:
this seems a bit weird to me. We will still hold the con lock if we crash anyways right?

There is no reason we should panic in this code, so it seems like this func() thing is an unnecessary safeguard?

Reply:
I have covered the only known source of panics.  This is guarding against the unknown ones...